packages/backend-core/src/logging/pino/logger.ts

```diff
@@ -10,6 +10,18 @@ import { formats } from "dd-trace/ext"
 
 import { localFileDestination } from "../system"
 
+function isPlainObject(obj: any) {
```

Comment:
Should we keep them inner as `const`?

Reply:
Because these were pure functions that didn't depend on the surrounding context, I opted to put them at the top level. Don't particularly mind either way, but if there's no need for them to be inner I lean slightly toward them not being inner.